libr/core/cmd.c

```diff
@@ -3111,31 +3131,30 @@ static int compare_cmd_descriptor_name(const void *a, const void *b) {
 }
 
 static void cmd_descriptor_init(RCore *core) {
-#if 0
-	// All this code is crap, buggy, slow and incomplete, also is making r2 crash
-	// commenting out for now
-	qsort (cmd_descriptors, cmd_descriptors_len, sizeof (RCmdDescriptor), compare_cmd_descriptor_name);
-	int i, n_cmd_descriptors = cmd_descriptors_len;
-	for (i = 0; i < n_cmd_descriptors; i++) {
-		const ut8 *p;
-		RCmdDescriptor *x = &core->cmd_descriptor;
-		for (p = (const ut8*)cmd_descriptors[i].name; *p; p++) {
+	const ut8 *p;
+	RListIter *iter;
+	RCmdDescriptor *x, *y;
+	int n = core->cmd_descriptors->length;
+	r_list_sort (core->cmd_descriptors, compare_cmd_descriptor_name);
+	r_list_foreach (core->cmd_descriptors, iter, y) {
+		if (--n < 0) break;
+		x = &core->root_cmd_descriptor;
+		for (p = (const ut8 *)y->name; *p; p++) {
 			if (!x->sub[*p]) {
-				if (cmd_descriptors_len >= MAX_NUM_CMD_DESCRIPTORS) {
-					eprintf ("Please increase MAX_NUM_CMD_DESCRIPTORS in libr/core/cmd.c\n");
-					exit(1);
+				if (p[1]) {
+					RCmdDescriptor *d = R_NEW0 (RCmdDescriptor);
```

Comment:
`r_list_append` does not append the element if `data` if NULL. So, the null case is already handled.